core/src/main/web/app/mainapp/mainapp.js

```diff
@@ -541,8 +541,53 @@
             }
           };
 
+          function _closeNotebook() {
+            var self = this;
+            var closeSession = function() {
+              bkSessionManager.close().then(function() {
+                bkCoreManager.gotoControlPanel();
+              });
+            };
+            if (bkSessionManager.isNotebookModelEdited() === false) {
+              closeSession();
+            } else {
+              var notebookTitle = bkSessionManager.getNotebookTitle();
+              bkHelper.show3ButtonModal(
+                  "Do you want to save " + notebookTitle + "?",
+                  "Confirm close",
+                  function() {
+                    self.saveNotebook().then(closeSession);
```

Comment:
on close it is going to call something async, what if the save fails? what is someone starts another save or async process and then the notebook closes? could be bad right?